### PR TITLE
Collect tslibs test, organize PeriodIndex tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Parametrize a bunch of nested tests.

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_arithmetic.py [hunk 1]
```diff
@@ -933,3 +935,16 @@ def test_datetime64_with_DateOffset(klass, assert_func):
                  Timestamp('2000-02-29', tz='US/Central')], name='a')
     assert_func(result, exp)
     assert_func(result2, exp)
+
+
+@pytest.mark.parametrize('years,months', product([-1, 0, 1], [-2, 0, 2]))
```
Comment (maxim-lian) [inline pandas/tests/indexes/datetimes/test_arithmetic.py]: If you parametrize years & months separately, pytest will handle the `product` 

i.e. `@pytest.mark.parametrize(('years',[-1, 0, 1]), ('months', [-2, 0, 2]))`

File: pandas/tests/indexes/datetimes/test_astype.py [hunk 2]
```diff
@@ -138,6 +138,48 @@ def test_astype_object(self):
         tm.assert_index_equal(casted, Index(exp_values, dtype=np.object_))
         assert casted.tolist() == exp_values
 
+        idx = pd.date_range(start='2013-01-01', periods=4, freq='M',
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_astype.py]: can you avoid duplication of lots of code and just parameterize over the tz.

File: pandas/tests/indexes/datetimes/test_tools.py [hunk 3]
```diff
@@ -1745,3 +1577,15 @@ def test_processing_order(self):
         result = pd.to_datetime(300 * 365, unit='D', origin='1870-01-01')
         expected = Timestamp('2169-10-20 00:00:00')
         assert result == expected
+
+
+def test_to_datetime1():
+    actual = pd.to_datetime(datetime(2008, 1, 15))
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_tools.py]: can you add a refernce for this test (e.g. what is it testing)

can you put it near other tests that are similar

File: pandas/tests/indexes/period/test_arithmetic.py [hunk 4]
```diff
@@ -250,6 +250,79 @@ def test_sub_isub(self):
         rng -= 1
         tm.assert_index_equal(rng, expected)
 
+    # ------------------------------------------------------------------
+    # PeriodIndex.shift is used by __add__ and __sub__
```
Comment (jreback) [inline pandas/tests/indexes/period/test_arithmetic.py]: can you parameterize

File: pandas/tests/tslibs/test_period_helper.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        assert period_asfreq(1, get_freq('S'),
+                             get_freq('N'), False) == 1000000000
+
+        assert period_asfreq(1, get_freq('L'), get_freq('U'), False) == 1000
+        assert period_asfreq(1, get_freq('L'),
+                             get_freq('N'), False) == 1000000
+
+        assert period_asfreq(1, get_freq('U'), get_freq('N'), False) == 1000
+
+    def test_period_ordinal_start_values(self):
```
Comment (jreback) [inline pandas/tests/tslibs/test_period_helper.py]: parameterize

File: pandas/tests/tslibs/test_period_helper.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        # information for 1.1.1970
+        assert period_ordinal(1970, 1, 1, 0, 0, 0, 0, 0, get_freq('A')) == 0
```
Comment (jreback) [inline pandas/tests/tslibs/test_period_helper.py]: same

File: pandas/tests/tslibs/test_period_helper.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        assert period_ordinal(1970, 1, 1, 0, 0, 0, 0, 0, get_freq('M')) == 0
+        assert period_ordinal(1970, 1, 1, 0, 0, 0, 0, 0, get_freq('W')) == 1
+        assert period_ordinal(1970, 1, 1, 0, 0, 0, 0, 0, get_freq('D')) == 0
+        assert period_ordinal(1970, 1, 1, 0, 0, 0, 0, 0, get_freq('B')) == 0
+
+    def test_period_ordinal_week(self):
+        assert period_ordinal(1970, 1, 4, 0, 0, 0, 0, 0, get_freq('W')) == 1
+        assert period_ordinal(1970, 1, 5, 0, 0, 0, 0, 0, get_freq('W')) == 2
```
Comment (jreback) [inline pandas/tests/tslibs/test_period_helper.py]: same

File: pandas/tests/tslibs/test_tslib.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    result = tslib.normalize_date(value)
+    assert (result == datetime(2012, 9, 7))
+
+    value = datetime(2012, 9, 7, 12)
+
+    result = tslib.normalize_date(value)
+    assert (result == datetime(2012, 9, 7))
+
+
+class TestParseISO8601(object):
```
Comment (jreback) [inline pandas/tests/tslibs/test_tslib.py]: these are conversion tests no? (so separate file). I don't really want much in tslib/test_testlib.py except if it doesnt' fit anywhere else

**Review by jreback — CHANGES_REQUESTED** [review]
lots of nice changes. but pls split into smaller prs. move by type (timedelta, period, timestamp & parsing)

**Comment by jreback** [comment]
see comments above

**Pull request closed without merging**